bagitobjecttransfer/recordtransfer/tests/unit/test_forms.py

```diff
@@ -433,3 +434,79 @@ def test_form_no_files_uploaded(self) -> None:
         form = UploadFilesForm(data=form_data)
         self.assertFalse(form.is_valid())
         self.assertIn("session_token", form.errors)
+
+
+class SubmissionGroupFormTest(TestCase):
+    """Tests for the SubmissionGroupForm."""
+
+    def setUp(self) -> None:
+        """Create initial test data."""
+        self.user = User.objects.create_user(username="testuser", password="12345")
+        self.existing_group = SubmissionGroup.objects.create(
+            name="Existing Group", description="An existing submission group", created_by=self.user
+        )
+
+    def test_form_valid_data(self) -> None:
+        """Case where the form is valid."""
+        form_data = {"name": "New Group", "description": "A new submission group"}
+        form = SubmissionGroupForm(data=form_data, user=self.user)
+        assert form.is_valid()
+        group = form.save()
+        assert group.name == "New Group"
+        assert group.description == "A new submission group"
+        assert group.created_by == self.user
+
+    def test_form_empty_data(self) -> None:
+        """Case where the form is empty."""
+        form = SubmissionGroupForm(data={}, user=self.user)
+        assert not form.is_valid()
+
+    def test_form_missing_name(self) -> None:
+        """Case where the name is missing."""
+        form_data = {"description": "A new submission group"}
+        form = SubmissionGroupForm(data=form_data, user=self.user)
+        assert not form.is_valid()
+        assert "name" in form.errors
+
+    def test_form_duplicate_name(self) -> None:
+        """Case where the name is a duplicate of an existing group name."""
+        form_data = {"name": "Existing Group", "description": "A duplicate submission group"}
+        form = SubmissionGroupForm(data=form_data, user=self.user)
+        assert not form.is_valid()
+        assert "name" in form.errors
+
+    def test_form_no_changes(self) -> None:
+        """Case where no changes are detected."""
+        form_data = {"name": "Existing Group", "description": "A new description"}
+        form = SubmissionGroupForm(data=form_data, user=self.user)
+        form.is_valid()  # Trigger the clean method
+        assert not form.is_valid()
```

Comment:
I believe the first `is_valid()` is redundant. If it's not, can you add a slightly more detailed comment as to why it's needed?

```suggestion
        self.assertFalse(form.is_valid())
```

Reply:
Yes, the first one is redundant. I've fixed that here:
https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/347/commits/d11d93f3b65df1e19fa36a42c4cc0cf624b3929f